src/wazuh_modules/wm_control.c

```diff
@@ -80,32 +80,36 @@ char* getPrimaryIP(){
 #if defined __linux__ || defined __MACH__
     cJSON *object;
     if (sysinfo_network_ptr && sysinfo_free_result_ptr) {
-        sysinfo_network_ptr(&object);
-        if (object) {
-            const cJSON *iface = cJSON_GetObjectItem(object, "iface");
-            if (iface) {
-                const int size_ids = cJSON_GetArraySize(iface);
-                for (int i = 0; i < size_ids; i++){
-                    const cJSON *element = cJSON_GetArrayItem(iface, i);
-                    if(!element) {
-                        continue;
-                    }
-                    cJSON *gateway = cJSON_GetObjectItem(element, "gateway");
-                    if(gateway && cJSON_GetStringValue(gateway) && 0 != strcmp(gateway->valuestring,"unkwown")) {
-                        const cJSON *ipv4 = cJSON_GetObjectItem(element, "IPv4");
-                        if (!ipv4) {
+        if (sysinfo_network_ptr(&object) != -1) {
+            if (object) {
+                const cJSON *iface = cJSON_GetObjectItem(object, "iface");
+                if (iface) {
+                    const int size_ids = cJSON_GetArraySize(iface);
+                    for (int i = 0; i < size_ids; i++){
+                        const cJSON *element = cJSON_GetArrayItem(iface, i);
+                        if(!element) {
                             continue;
                         }
-                        cJSON *address = cJSON_GetObjectItem(ipv4, "address");
-                        if (address && cJSON_GetStringValue(address))
-                        {
-                            os_strdup(address->valuestring, agent_ip);
-                            break;
+                        cJSON *gateway = cJSON_GetObjectItem(element, "gateway");
+                        if(gateway && cJSON_GetStringValue(gateway) && 0 != strcmp(gateway->valuestring,"unkwown")) {
+                            const cJSON *ipv4 = cJSON_GetObjectItem(element, "IPv4");
+                            if (!ipv4) {
+                                continue;
+                            }
+                            cJSON *address = cJSON_GetObjectItem(ipv4, "address");
+                            if (address && cJSON_GetStringValue(address))
+                            {
+                                os_strdup(address->valuestring, agent_ip);
+                                break;
+                            }
                         }
                     }
                 }
+                sysinfo_free_result_ptr(&object);
             }
-            sysinfo_free_result_ptr(&object);
+        }
+        else {
+            mterror(WM_CONTROL_LOGTAG, "Unable to get system network information (%d) %s.", errno, strerror(errno));
```

Comment:
return code should be used here instead of errno

Reply:
Will do